OpenKeychain/src/main/java/org/sufficientlysecure/keychain/keyimport/WebKeyDirectoryClient.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2017 Schürmann & Breitmoser GbR
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.sufficientlysecure.keychain.keyimport;
+
+
+import android.support.annotation.NonNull;
+
+import org.sufficientlysecure.keychain.network.OkHttpClientFactory;
+import org.sufficientlysecure.keychain.pgp.UncachedKeyRing;
+import org.sufficientlysecure.keychain.pgp.UncachedPublicKey;
+import org.sufficientlysecure.keychain.pgp.exception.PgpGeneralException;
+import org.sufficientlysecure.keychain.ui.util.KeyFormattingUtils;
+import org.sufficientlysecure.keychain.util.ParcelableProxy;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.net.URL;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import timber.log.Timber;
+
+
+public class WebKeyDirectoryClient implements KeyserverClient {
+
+    public static WebKeyDirectoryClient getInstance() {
+        return new WebKeyDirectoryClient();
+    }
+
+    private WebKeyDirectoryClient() { }
+
+    private static Pattern EMAIL_PATTERN = Pattern.compile("^\\s*(.+)@(.+)\\s*$");
```

Comment:
regex cannot validate 100% an email address.

see here for a correct implementatio (apache commons) https://github.com/apache/commons-validator/blob/trunk/src/main/java/org/apache/commons/validator/EmailValidator.java

Reply:
@MauroMombelli thanks, unfortunately it doesn't seem to have extraction of local/domain part and I need that.
My point is if it's really necessary to actually validate e-mail here, the e-mail may be valid but using some obscure things like comments etc. that I would need to strip out. So it could be a lot of code (more code -> more bugs) for a (probably) marginal benefit.

Reply:
I think it's fine. Worse case it'll die in dns resolution or request a nonexisting hash :)